pref-adjust.lua

```diff
@@ -91,16 +136,201 @@ function clearpref_all_dwarves()
+    -- <material 0:1 INORGANIC:GOLD>
+    for k,v in ipairs(df.item_type) do
+        list_of_items[v]=k
+        list_of_items_string=list_of_items_string..v..","
+    end
+    if printflag==1 then
+        print("\nTYPE 4 ITEMS:"..list_of_items_string) --    printall(list_of_items)
+    end
+-- ------------------------------------
+    list_of_colors={} -- Type 7 "Likes the color.."
+    list_of_colors_string = ""
+    vec=df.global.world.raws.descriptors.colors
+    for k=0,#vec-1 do
+        name=vec[k].id
+        list_of_colors[name]=k
+        list_of_colors_string=list_of_colors_string..name..","
+    end
+    if printflag==1 then
+        print("\nTYPE 7 COLORS:"..list_of_colors_string) --    printall(list_of_colors)
+    end
+-- ------------------------------------
+    list_of_shapes={} -- Type 8 "Likes circles"
+    list_of_shapes_string = ""
+    vec=df.global.world.raws.descriptors.shapes
+    for k=0,#vec-1 do
+        name=vec[k].id
+        list_of_shapes[name]=k
+        list_of_shapes_string=list_of_shapes_string..name..","
+    end
+    if printflag==1 then
+        print("\nTYPE 8 SHAPES:"..list_of_shapes_string) --    printall(list_of_shapes)
+    end
+-- ------------------------------------
+    list_of_poems={} -- Type 9 "likes the words of.."
+    list_of_poems_string = ""
+    vec=df.global.world.poetic_forms.all
+    for k=0,#vec-1 do
+        name=dfhack.TranslateName(vec[k].name,true)
+        list_of_poems[name]=k
+        list_of_poems_string=list_of_poems_string..k..":"..name..","
+    end
+    if printflag==1 then
+        print("\nTYPE 9 POEMS:"..list_of_poems_string) -- printall(list_of_poems)
+    end
+-- ------------------------------------
+    list_of_music={} -- Type 10 "Likes the sound of.."
+    list_of_music_string = ""
+    vec=df.global.world.musical_forms.all
+    for k=0,#vec-1 do
+        name=dfhack.TranslateName(vec[k].name,true)
+        list_of_music[name]=k
+        list_of_music_string=list_of_music_string..k..":"..name..","
+    end
+    if printflag==1 then
+        print("\nTYPE 10 MUSIC:"..list_of_music_string) --    printall(list_of_music)
+    end
+-- ------------------------------------
+    list_of_dances={} -- Type 11
+    list_of_dances_string = ""
+    vec=df.global.world.dance_forms.all
+    for k=0,#vec-1 do
+        name=dfhack.TranslateName(vec[k].name,true)
+        list_of_dances[name]=k
+        list_of_dances_string=list_of_dances_string..k..":"..name..","
+    end
+    if printflag==1 then
+        print("\nTYPE 11 DANCES:"..list_of_dances_string) --    printall(list_of_dances)
+    end
+end -- end func build_all_lists
+
 -- ---------------------------------------------------------------------------
 -- main script operation starts here
 -- ---------------------------------------------------------------------------
-clearpref_all_dwarves()
-adjust_all_dwarves()
+pss_counter=31415926
+utils = require 'utils'
+printflag=0
+build_all_lists(printflag)
+
+local opt = ...
```

Comment:
This is not obvious that it assigns only the *first* argument to the script to `opt`. Something like this, albeit ugly, is more explicit:
```lua
local opt = ({...})[1]
```
or alternatively:
```lua
select(1, ...)
```
(the last one is probably better)